src/Compilers/Server/VBCSCompilerTests/CompilerServerTests.cs

```diff
@@ -143,39 +98,117 @@ private static void CheckForBadShared(string arguments)
-            IEnumerable<KeyValuePair<string, string>> filesInDirectory,
-            IEnumerable<KeyValuePair<string, string>> additionalEnvironmentVars = null)
+        private static void CreateFiles(TempDirectory currentDirectory, IEnumerable<KeyValuePair<string, string>> files)
         {
-            CheckForBadShared(arguments);
-            foreach (var pair in filesInDirectory)
+            if (files != null)
             {
-                TempFile file = currentDirectory.CreateFile(pair.Key);
-                file.WriteAllText(pair.Value);
+                foreach (var pair in files)
+                {
+                    TempFile file = currentDirectory.CreateFile(pair.Key);
+                    file.WriteAllText(pair.Value);
+                }
             }
+        }
 
-            return RunCommandLineCompiler(
-                compilerPath,
-                arguments,
-                currentDirectory.Path,
-                additionalEnvironmentVars: AddForLoggingEnvironmentVars(additionalEnvironmentVars));
+        private static T ApplyEnvironmentVariables<T>(
+            IEnumerable<KeyValuePair<string, string>> environmentVariables,
+            Func<T> func)
+        {
+            if (environmentVariables == null)
+            {
+                return func();
+            }
+            var resetVariables = new Dictionary<string, string>();
+            try
+            {
+                foreach (var variable in environmentVariables)
+                {
+                    resetVariables.Add(variable.Key, Environment.GetEnvironmentVariable(variable.Key));
+                    Environment.SetEnvironmentVariable(variable.Key, variable.Value);
+                }
+                return func();
+            }
+            finally
+            {
+                foreach (var variable in resetVariables)
+                {
+                    Environment.SetEnvironmentVariable(variable.Key, variable.Value);
+                }
+            }
+        }
+
+        private static (T result, string output) UseTextWriter<T>(Encoding encoding, Func<TextWriter, T> func)
+        {
+            MemoryStream memoryStream;
+            TextWriter writer;
+            if (encoding == null)
+            {
+                memoryStream = null;
+                writer = new StringWriter();
+            }
+            else
+            {
+                memoryStream = new MemoryStream();
+                writer = new StreamWriter(memoryStream, encoding);
+            }
+            var result = func(writer);
+            writer.Flush();
+            if (memoryStream != null)
+            {
+                return (result, encoding.GetString(memoryStream.GetBuffer(), 0, (int)memoryStream.Length));
+            }
+            else
+            {
+                return (result, ((StringWriter)writer).ToString());
+            }
+        }
+
+        internal static (int exitCode, string output) RunCommandLineCompiler(
```

Comment:
Consider naming the fields `(int ExitCode, string Output)`, for consistency with previous code, and because public fields typically start with an uppercase character.

Reply:
Jared specifically told me to lowercase these - so I'm not sure what to do.

Reply:
Let's use uppercase for now, to minimize the differences with the current code, and we can change later in a separate PR if necessary.